lib/tool_shed/util/shed_util_common.py

```diff
@@ -512,6 +512,11 @@ def get_repository_for_dependency_relationship( app, tool_shed, name, owner, cha
                                                owner=owner,
                                                changeset_revision=changeset_revision )
     if not repository:
+        tool_shed_url = common_util.get_tool_shed_url_from_tool_shed_registry( app, tool_shed )
+        repository_clone_url = os.path.join( tool_shed_url, 'repos', owner, name )
+        repo_info_tuple = (None, repository_clone_url, changeset_revision, None, owner, None, None)
+        repository, pcr = repository_was_previously_installed( app, tool_shed_url, name, repo_info_tuple )
```

Comment:
@nsoranzo You make a good point, but the ideal solution to this is to rebase these functions and clean them up, which may be a bit non-trivial.  My understanding is that all of this is going away some time soon?  

I'm going to try to get some time to look at the 2nd problem discussed here (unless someone else gets to it first) https://github.com/galaxyproject/galaxy/issues/667#issuecomment-159305155
If it makes sense to clean this function up a bit, I'll do it as part of that work.

By the way, what Tool Shed test harness should I be running to test PRs like this?  I want to make sure I'm not breaking things.  ;)

Thanks!